modules/liftover/1.0/liftover.smk

```diff
@@ -23,46 +23,35 @@ CFG = op.setup_module(
     version = "1.0",
     subdirectories = ["inputs", "seghg38tobedhg38", "bedhg38tobedhg19", "bedhg19toseghg19", "outputs"])
 
-# Define rules to be run locally when using a compute cluster
-localrules:
-    _liftover_input_seg,
-    _hg38seg_2_hg38bed,
-    _hg38bed_2_hg19bed,
-    _hg19bed_2_hg19seg,
-    _liftover_output_seg,
-    _liftover_all,
-
 
 ##### RULES #####
 
 # Symlinks the input files into the module results directory (under '00-inputs/')
 rule _liftover_input_seg:
     input:
-        seg = CFG["inputs"]["sample_seg"]
+        seg = expand(CFG["inputs"]["sample_seg"], tumour_sample_id=CFG["runs"]["tumour_sample_id"], normal_sample_id=CFG["runs"]["normal_sample_id"], tool=CFG["tool"])
```

Comment:
If the user is setting this tool as "battenberg" (as per the documentation in the config) then how will this ever match the file naming produced by battenberg? Is the user expected to rename all those files first?

Reply:
No, there will be no need to rename the files before using with this module. I specified these wildcards so they will be recognized from input files, otherwise, snakemake gives an error that it cannot determine wildcards of output files from the inputs. I tested it with the file input named 08-20595T_subclones.hg38.igv.seg, which will be the output of battenberg

Reply:
I don't understand. 
tool: __UPDATE__ # Specify the name of the tool that produced the .seg files. For example, "battenberg" 

Where does the value of this variable get used and how does it allow the input files to be recognized? I must be missing something here.

Reply:
It will be used to retain the name of the tool that produced seg files in the naming of final files that are produced by liftover. It is used as a wildcard {tool} in the liftover.smk file, and it is the sole purpose of this config option. I used it so when there are final lifted files, it will be possible to recognize which CNV caller was used producing original seg files. It will allow the input files to be recognized as belonging to a certain tool. Instead, in the config file, both "tool" and "sample_seg" needs to be updated.

Reply:
input_seg: "results/gambl/{tool}_0.0/{genome_build}/{tumour_sample_id}--{normal_sample_id}/{tumour_sample_id}_subclones_1.igv.seg"

Isn't this what your config would look like?

Reply:
> To clarify, it seems to me this would only work if you started with input files that are named something along these lines:
> "seg/20595T--20595N.battenberg.hg38.igv.seg"

I got it now. No, it will work based on what is in the config. I am reading the file from what is specified in config, regardless of its naming pattern - what the user provides in the filed "sample_seg" of the input in the config/default.yaml . But, within this rule and during symlinking, the output (the symlink in the 00-inputs folder) will have a different naming pattern - this is where I am taking advantage of bringing everything to the same consistent naming pattern. Now that I am writing this I also realized that the sample will be recognized only if it is listed in the samples table, and only if it is paired.

Reply:
> input_seg: "results/gambl/{tool}_0.0/{genome_build}/{tumour_sample_id}--{normal_sample_id}/{tumour_sample_id}_subclones_1.igv.seg"
> 
> Isn't this what your config would look like?

let me test a path like this and naming scheme like this

Reply:
shouldn't it be with 99-outputs?
`input_seg: "results/gambl/{tool}_0.0/99-outputs/{genome_build}/{tumour_sample_id}--{normal_sample_id}/{tumour_sample_id}_subclones_1.igv.seg"`
I tested with the path you mentioned and it does not recognize wild card genome build in the configuration I pushed. I thought this wildcard is "build-in" into lcr-modules and I would not need to specify it. If I specify the genome_build wildcard, it is working. This is the output for the first rule:
```
rule _liftover_input_seg:
    input: results/gambl/battenberg_0.0/hg38/08-20595T--08-20595N/08-20595T_subclones_1.igv.seg
    output: results/liftover-1.0/00-inputs/seg/08-20595T--08-20595N.battenberg.igv.seg
    jobid: 5
    reason: Missing output files: results/liftover-1.0/00-inputs/seg/08-20595T--08-20595N.battenberg.igv.seg
    wildcards: tumour_sample_id=08-20595T, normal_sample_id=08-20595N, tool=battenberg
```